File: dbilo-siuntos-backend/Infrastructure/Models/Parcel.cs
using Infrastructure.Enums;
using System.ComponentModel.DataAnnotations;

namespace Infrastructure.Models
{
    public class Parcel
    {
        public int Id { get; set; }
        [Key]
        public string TrackingNumber { get; set; }
        public ParcelSize Size { get; set; }

        public RecipientDetails ShipperDetails { get; set; }
        public RecipientDetails ReceiverDetails { get; set; }

        public User? Shipper { get; set; }

        public Address? PickupAddress { get; set; }
        public Address? DeliveryAddress { get; set; }

        public ICollection<Status> Status { get; set; }

        public Terminal? PickupTerminal { get; set; }
        public Terminal? DeliveryTerminal { get; set; }

        [Timestamp]
        public byte[] RowVersion { get; set; }
    }
}

File: dbilo-siuntos-frontend/src/types/parcel.ts
import { Address } from './address';
import { ParcelSize } from './parcelSize';
import { RecipientDetails } from './recipientDetails';
import { Status } from './status';
import { Terminal } from './terminal';
import { User } from './user';

export interface Parcel {
  id: number;
  trackingNumber: string;
  size: ParcelSize;
  shipperDetails: RecipientDetails;
  receiverDetails: RecipientDetails;
  shipper?: User;
  pickupAddress?: Address;
  deliveryAddress?: Address;
  status: Status[];
  pickupTerminal?: Terminal;
  deliveryTerminal?: Terminal;
  rowVersion: string;
}
